src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+                    do_write(1);
+                }
+
+                if (status & ACCOUNT_STATUS_CODE::NORMAL)
+                {
+                    // Account info verified, grab password
+                    std::string updated_password(data_ + 0x30, 16);
+
+                    if (updated_password == "")
+                    {
+                        ShowWarning(fmt::format("login_parse: Empty password; Could not update password for user <{}>.", escaped_name));
+                        ref<uint8>(data_, 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                        do_write(1);
+                        return;
+                    }
+
+                    char escaped_updated_password[16 * 2 + 1];
+                    sql->EscapeString(escaped_updated_password, updated_password.c_str());
+
+                    fmtQuery = "UPDATE accounts SET accounts.timelastmodify = NULL WHERE accounts.id = %d";
+                    sql->Query(fmtQuery, accid);
+
+                    fmtQuery = "UPDATE accounts SET accounts.password = PASSWORD('%s') WHERE accounts.id = %d";
+                    ret      = sql->Query(fmtQuery, escaped_updated_password, accid);
+                    if (ret == SQL_ERROR)
+                    {
+                        ShowWarning(fmt::format("login_parse: Error trying to update password in database for user <{}>.", escaped_name));
+                        ref<uint8>(data_, 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                        do_write(1);
+                        return;
+                    }
+
+                    memset(data_, 0, 33);
+                    ref<uint8>(data_, 0) = LOGIN_SUCCESS_CHANGE_PASSWORD;
+                    do_write(33);
+
+                    ShowInfo("login_parse: password updated successfully.");
+                    return;
+                }
+            }
+            break;
+            default:
+            {
+                ShowError(fmt::format("Unhandled auth code: {}", code));
+            }
+            break;
+        }
+    }
+
+    void write_func() override
+    {
+        do_read();
+    }
+
+    void handle_error(std::error_code ec, std::shared_ptr<handler_session> self) override
+    {
+    }
+
+    void do_write(std::size_t length)
+    {
+        auto self(shared_from_this());
+        // clang-format off
+        asio::async_write(socket_, asio::buffer(data_, length),
+        [this, self](std::error_code ec, std::size_t /*length*/)
+        {
+            if (!ec)
+            {
+                write_func();
+            }
+            else
+            {
+                ShowError(ec.message());
+            }
+        });
+        // clang-format on
+    }
+};
+
+void PrintPacket(const char* data, uint32 size)
+{
+    std::string message;
+    char        buffer[5];
+
+    for (size_t y = 0; y < size; y++)
+    {
+        std::memset(buffer, 0, sizeof(buffer));               // TODO: Replace these three lines with std::format when/if we move to C++ 20.
```

Comment:
God I hate this, can you try and use fmt lib/spdlog to kill this?

Reply:
Killed as requested